### PR TITLE
Linted generated migration files, join table improvements

### PR DESCRIPTION
**Pull request by wswoodruff**
- Lotta work to decide where the semicolons go for generated linting
- Added rollback testing to ensure the exports.down code works as expected

### CODE REVIEW

File: test/utils/index.js [hunk 1]
```diff
@@ -5,6 +5,55 @@ const Path = require('path');
...
+            session.options.knexConfig,
+            { directory: rollbackPath }
+        );
+
+        knex.migrate.currentVersion()
+            .then((cv) => {
+
+                if (cv !== 'none') {
+                    return knex.migrate.rollback(config)
+                        .then(() => {
```
Comment (devinivy) [inline test/utils/index.js]: Knex promises should have an `asCallback()` method.

File: .eslintignore [hunk 2]
```diff
@@ -1,5 +1,9 @@
 node_modules/*
+
+/* If you remove this next line, the tests will fail with
```
Comment (devinivy) [inline .eslintignore]: I don't believe eslintignore supports `/* */` style comments (but does support `#`).

Comment (devinivy) [inline .eslintignore]: What is the reason for this side-effect?  Do we understand it, and can we avoid it some other way?  I think we would like to lint the generated migrations ideally (although that could be done separately in its own test).

File: lib/mappings.js [hunk 3]
```diff
@@ -103,10 +112,18 @@ const convertFuncs = {
 
         }, {});
 
-        Hoek.assert(badChildren.length === 0, `Joi Schema type(s) "${badChildren}" not supported.`);
+        Hoek.assert(badChildren.length === 0, `Joi Schema type(s) "${badChildren}" not supported. Please fix mappings.js`);
```
Comment (devinivy) [inline lib/mappings.js]: Is this a user-facing error?  I wonder if `Please fix mappings.js` could be confusing to a user.

**Comment by wswoodruff** [comment]
Closing in favor of #7

**Pull request closed without merging**